state/unit.go

```diff
@@ -95,14 +96,18 @@ type unitDoc struct {
 
 // Unit represents the state of a service unit.
 type Unit struct {
-	st  *State
-	doc unitDoc
+	storagePath string
+	st          *State
+	doc         unitDoc
 }
 
 func newUnit(st *State, udoc *unitDoc) *Unit {
 	unit := &Unit{
-		st:  st,
-		doc: *udoc,
+		// TODO(katco): Pass this in; unsure if I should store this in
+		// Mongo or not.
+		storagePath: paths.Defaults.Storage,
```

Comment:
This is certainly wrong as we could end up using the local machines OS to determine the storage path of a remote unit. To avoid the possibility of errors shouldn't the storage path always be calculated using the OS of the host machine?

I don't think this should be stored in mongo, nor passed in (at this level at least).